packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,104 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type { AvailableUpdateInfo, UpdatesInfo, UseUpdatesCallbacksType } from './UseUpdates.types';
+import { currentlyRunning } from './UseUpdatesConstants';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+// Manifest is of type "any" until we no longer support classic updates
+export const availableUpdateFromManifest = (manifest: any) => {
```

Comment:
(this is `Manifest` from `Updates.types`)

```suggestion
export const availableUpdateFromManifest = (manifest: Manifest) => {
```

To do this, we should change a few things:
```
export type ClassicManifest = NonNullable<typeof Constants.manifest>;

export type Manifest = ClassicManifest | NonNullable<typeof Constants.manifest2>;
```

```
export type UpdateEvent =
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.ERROR;
      /**
       * If `type` is `Updates.UpdateEventType.ERROR`, the error message, and `undefined` otherwise.
       */
      message: string;
    }
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.UPDATE_AVAILABLE;
      /**
       * If `type` is `Updates.UpdateEventType.UPDATE_AVAILABLE`, the manifest of the newly downloaded
       * update, and `undefined` otherwise.
       */
      manifest: Manifest;
    }
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.NO_UPDATE_AVAILABLE;
    };
```

Reply:
Done